src/coreclr/jit/valuenumfuncs.h

```diff
@@ -143,7 +143,8 @@ ValueNumFuncDef(JitReadyToRunNewArr, 3, false, true, false)
 ValueNumFuncDef(Box, 3, false, false, false)
 ValueNumFuncDef(BoxNullable, 3, false, false, false)
 
-ValueNumFuncDef(StrCns, 2, false, true, false)
+ValueNumFuncDef(LazyStrCns, 2, false, true, false)  // lazy-initialized string literal (helper)
```

Comment:
Could you please explain why we need a separate func here?

Reply:
They both accept integers as arguments but in the case of StrCns it's "address of the string object (indirect)" and in case of `LazyStrCns` it's `ScpHnd` in theory can have the same value but will produce a different string so I decided to create a new one.